test/lib/tour-overlay/TourOverlay.component.test.tsx

```diff
@@ -82,4 +88,103 @@ describe("[Integration] TourOverlay.component.test.tsx", () => {
       expect(queryByText("Step 2")).toBeNull();
     });
   });
+
+  context("when the backdrop behavior is set to continue", () => {
+    it("goes to the next step when the backdrop is pressed", async () => {
+      const { getByText, findByTestId, queryByText } = render(
+        <SpotlightTourProvider steps={STEPS} onBackdropPress="continue">
+          <TestScreen />
+        </SpotlightTourProvider>
+      );
+
+      await waitFor(() => getByText("Step 1"));
+
+      const backdrop = await findByTestId("Spot Svg");
+
+      fireEvent.press(backdrop);
+
+      await waitFor(() => getByText("Step 2"));
+
+      expect(queryByText("Step 1")).toBeNull();
+    });
+  });
+
+  context("when the backdrop behavior is set to stop", () => {
+    it("stops the tour when the backdrop is pressed", async () => {
+      const { getByText, findByTestId, queryByText } = render(
+        <SpotlightTourProvider steps={STEPS} onBackdropPress="stop">
+          <TestScreen />
+        </SpotlightTourProvider>
+      );
+
+      await waitFor(() => getByText("Step 1"));
+
+      const backdrop = await findByTestId("Spot Svg");
+
+      fireEvent.press(backdrop);
+
+      expect(queryByText("Step 1")).toBeNull();
+      expect(queryByText("Step 2")).toBeNull();
+      expect(queryByText("Step 3")).toBeNull();
+    });
+  });
+
+  context("when the backdrop behavior is present in the step", () => {
+    it("overrides the backdrop press default behavior", async () => {
+      const steps = STEPS.map<TourStep>((step, i) => {
+        return i === 1
+          ? { ...step, onBackdropPress: "stop" }
+          : step;
+      });
+
+      const { getByText, findByTestId, queryByText } = render(
+        <SpotlightTourProvider steps={steps} onBackdropPress="continue">
+          <TestScreen />
+        </SpotlightTourProvider>
+      );
+
+      await waitFor(() => getByText("Step 1"));
+
+      const backdrop = await findByTestId("Spot Svg");
+
+      fireEvent.press(backdrop);
+
+      await waitFor(() => getByText("Step 2"));
+
+      expect(queryByText("Step 1")).toBeNull();
+
+      fireEvent.press(backdrop);
+
+      expect(queryByText("Step 1")).toBeNull();
+      expect(queryByText("Step 2")).toBeNull();
+      expect(queryByText("Step 3")).toBeNull();
+    });
+  });
+
+  context("when a function is passed to the backdrop press behavior", () => {
+    it("injects the SpotlightTour object in the options", async () => {
+      const spy = Sinon.spy<(options: SpotlightTour) => void>(() => undefined);
```

Comment:
Is it not possible to use `jest.fn` instead as a `spy` to avoid the necessity of installing Sinon? If I remember correctly it can be typed too, but not sure 🤔

Reply:
The types of Jest are heavily based on `any`, so most of the time, they are not so safe to use. However, the main reason I prefer Sinon over Jest is that Sinon is framework agnostic. So if we decide to change the test framework, the tests will not break 🙂